src/main/java/org/jabref/logic/citationkeypattern/AbstractCitationKeyPattern.java

```diff
@@ -33,12 +32,19 @@ public static List<String> split(String bibtexKeyPattern) {
         // A holder for fields of the entry to be used for the key
         List<String> fieldList = new ArrayList<>();
 
-        // Before we do anything, we add the parameter to the ArrayLIst
+        // Before we do anything, we add the parameter to the ArrayList
         fieldList.add(bibtexKeyPattern);
 
-        StringTokenizer tok = new StringTokenizer(bibtexKeyPattern, "[]", true);
-        while (tok.hasMoreTokens()) {
-            fieldList.add(tok.nextToken());
+        // regex to split around a character
```

Comment:
While thinking more, the old code is much more readable. We also hit a perfect use case for StringTokenizer here:

![image](https://user-images.githubusercontent.com/1366654/229437359-187f37fe-1693-4617-a06a-d220163eec5f.png)